src/marqo/s2_inference/hf_utils.py

```diff
@@ -75,10 +164,46 @@ def forward(self, **kwargs):
     def mean_pooling(self, model_output, attention_mask):
 
         token_embeddings = model_output[0]
-        
+
         input_mask_expanded = attention_mask.unsqueeze(-1).expand(token_embeddings.size()).float()
 
         return torch.sum(token_embeddings * input_mask_expanded, 1) / torch.clamp(input_mask_expanded.sum(1), min=1e-9)
 
     def cls_pooling(self, model_output, attention_mask):
-        return model_output[0][:,0]
+        return model_output[0][:, 0]
+
+
+def validate_huggingface_archive(path: str) -> str:
+    '''
+        This function extracts the model from a huggingface archive if necessary.
+        Unlike open clip models that can loaded from a single .pt or .bin file,
+        Huggingface models are loaded from a directory.
+
+    Returns:
+        The directory path to the model
+    '''
+    if os.path.isdir(path):
+        # if it's a directory, return the path directly
+        return path
+    elif os.path.isfile(path):
+        # if it's a file, check if it's a compressed file
+        base, ext = os.path.splitext(path)
+        if ext in ['.tar', '.gz', '.tgz', '.bz2', '.zip']:
+            # create a new directory with the same name as the file
+            new_dir = base
+            os.makedirs(new_dir, exist_ok=True)
+
+            # extract the compressed file
+            if ext == '.zip':
+                with zipfile.ZipFile(path, 'r') as zip_ref:
+                    zip_ref.extractall(new_dir)
+            else:
+                with tarfile.open(path, 'r') as tar_ref:
+                    tar_ref.extractall(new_dir)
+
+            # return the path to the new directory
+            return new_dir
+        else:
+            raise InvalidModelPropertiesError(f'Unsupported file extension: {ext}. The path must be a directory or a compressed file.')
+    else:
```

Comment:
I don't think we need this (a path can either be a file or a directory, not a third option). Removing this means we can move :185 into a proper guard and simplify the code.